ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,301 @@
+}
+void op::v8::If::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v8_If_validate_and_infer_types);
+
+    NODE_VALIDATION_CHECK(
+        this, m_bodies.size() == 2, "If contains incorrect number of bodies:", m_bodies.size());
+
+    NODE_VALIDATION_CHECK(this,
+                          m_input_descriptions.size() == 2,
+                          "If contains incorrect number of body input descriptions:",
+                          m_input_descriptions.size());
+    NODE_VALIDATION_CHECK(this,
+                          m_output_descriptions.size() == 2,
+                          "If contains incorrect number of body output descriptions:",
+                          m_output_descriptions.size());
+
+    const auto& if_condition = input_value(0);
+    const auto& if_condition_rank = if_condition.get_partial_shape().rank();
+    if (if_condition_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              if_condition_rank.compatible(1) || if_condition_rank.compatible(0),
+                              "Rank of If condition input must be equal to 0 or 1");
+    }
+    // Trying to get cond as const value
+    if (const auto& cond_value = get_constant_from_source(if_condition))
+    {
+        auto val = cond_value->cast_vector<bool>();
+        NODE_VALIDATION_CHECK(
+            this,
+            val.size() == 1,
+            "The number of values in the If condition constant is greater than 1");
+
+        auto cond_index = val[0] ? then_body_index : else_body_index;
+        auto body = m_bodies[cond_index];
+        auto input_descriptors = m_input_descriptions[cond_index];
+        validate_and_infer_type_body(body, input_descriptors);
+        auto output_nodes = outputs();
+        for (const auto& output_descr : m_output_descriptions[cond_index])
+        {
+            auto body_value =
+                body->get_results().at(output_descr->m_body_value_index)->input_value(0);
+            auto body_value_partial_shape = body_value.get_partial_shape();
+            set_output_type(output_descr->m_output_index,
+                            body_value.get_element_type(),
+                            body_value_partial_shape);
+        }
+    }
+    else // condition is non constant
+    {
+        validate_and_infer_type_body(get_then_body(), m_input_descriptions[then_body_index]);
+        validate_and_infer_type_body(get_else_body(), m_input_descriptions[else_body_index]);
+        auto output_nodes = outputs();
+
+        auto then_outputs_map =
+            get_mapping_outputs_on_body_description(m_output_descriptions[then_body_index]);
+        auto else_outputs_map =
+            get_mapping_outputs_on_body_description(m_output_descriptions[else_body_index]);
+
+        for (size_t output_index = 0; output_index < output_nodes.size(); ++output_index)
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  then_outputs_map.count(output_index) != 0,
```

Comment:
I think that we need only one association always, otherwise, something goes wrong. So I propose: `then_outputs_map.count(output_index) == 1`